### ISSUE TITLE
SMB and Spotify Linking Errors

### ISSUE THREAD

**Issue by OzGav** [music-assistant/hass-music-assistant#1206]
### What version of Music Assistant has the issue?

2B29

### The problem

Local albums which have a Spotify equivalent are misbehaving in various ways. Firstly, please view this video: https://drive.google.com/file/d/1MSeZNzLmOCNOBQ9wYcVDTniGsqyhQOXT/view?usp=share_link

After what you saw in the video I then selected the album and played it and all the tracks are duplicated
![image](https://user-images.githubusercontent.com/19848947/233782120-959d271f-2f99-4d68-af38-b041ac27763b.png)

I also have this album. MA has put these tracks on the same album but has not linked the tracks. I'm assuming due to the word remaster in the title?
![image](https://user-images.githubusercontent.com/19848947/233782132-cd18a067-9bf0-453d-b674-739afd054316.png)
 

### How to reproduce

Have a local album with a Spotify equivalent where the tracks have a slightly different name.

### Additional information

What I expect to happen in the above cases is that the albums are not linked in the first place if all the tracks aren't going to link.  If the albums should be seen as the same then the tracks should also be linked.

### What version of Home Assistant Core are your running

2023.4.5

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by OzGav** [comment]
We had issues such as these on MA V1 that I worked with Marcel to eliminate. I believe it is difficult due to the lack of metadata supplied by Spotify. Then it is not helped by Spotify having artist name variations such as the two you indicated. Lets see what Marcel thinks of the above. There are a few weird problems like i the video I made where going to different parts of MA shows different information when it should be the same.
